File: 2025-1/sesion-de-clase-09/SoftProg/SoftProgPersistencia/src/main/java/pe/edu/pucp/inf30/softprog/daoimpl/rrhh/CuentaUsuarioDAOImpl.java
package pe.edu.pucp.inf30.softprog.daoimpl.rrhh;

import java.sql.CallableStatement;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Types;
import pe.edu.pucp.inf30.softprog.dao.rrhh.ICuentaUsuarioDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.BaseDAOImpl;
import pe.edu.pucp.inf30.softprog.rrhh.model.CuentaUsuario;

public class CuentaUsuarioDAOImpl extends BaseDAOImpl<CuentaUsuario> implements ICuentaUsuarioDAO {    
    @Override
    protected CallableStatement comandoInsertar(Connection conn, CuentaUsuario cuentaUsuario) throws SQLException {
        String sql = "{CALL insertarCuentaUsuario(?, ?, ?, ?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setString("p_userName", cuentaUsuario.getUserName());
        cmd.setString("p_password", cuentaUsuario.getPassword());
        cmd.setBoolean("p_activo", cuentaUsuario.isActivo());
        cmd.registerOutParameter("p_id", Types.INTEGER);
        return cmd;
    }
    
    @Override
    protected CallableStatement comandoModificar(Connection conn, CuentaUsuario cuentaUsuario) throws SQLException {
        String sql = "{CALL modificarCuentaUsuario(?, ?, ?, ?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setString("p_userName", cuentaUsuario.getUserName());
        cmd.setString("p_password", cuentaUsuario.getPassword());
        cmd.setBoolean("p_activo", cuentaUsuario.isActivo());
        cmd.setInt("p_id", cuentaUsuario.getId());
        return cmd;
    }

    @Override
    protected CallableStatement comandoEliminar(Connection conn, int id) throws SQLException {
        String sql = "{CALL eliminarCuentaUsuario(?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt(1, id);
        return cmd;
    }

    @Override
    protected CallableStatement comandoBuscar(Connection conn, int id) throws SQLException {
        String sql = "{CALL buscarCuentaUsuarioPorId(?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt(1, id);
        return cmd;
    }

    @Override
    protected CallableStatement comandoListar(Connection conn) throws SQLException {
        String sql = "{CALL listarCuentaUsuarios()}";
        CallableStatement cmd = conn.prepareCall(sql);
        return cmd;
    }

    @Override
    protected CuentaUsuario mapearModelo(ResultSet rs) throws SQLException {
        CuentaUsuario cuentaUsuario = new CuentaUsuario();
        cuentaUsuario.setId(rs.getInt("id"));
        cuentaUsuario.setUserName(rs.getString("userName"));
        cuentaUsuario.setPassword(rs.getString("password"));
        cuentaUsuario.setActivo(rs.getBoolean("activo"));

        return cuentaUsuario;
    }
}

File: 2025-1/sesion-de-clase-14/SoftProgPersistencia/DAOImpl/RRHH/CuentaUsuarioDAOImpl.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PUCP.Edu.Pe.SoftProg.Modelo.RRHH;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAO.RRHH;

namespace PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.RRHH {
    public class CuentaUsuarioDAOImpl : BaseDAOImpl<CuentaUsuario>, ICuentaUsuarioDAO {
        protected override DbCommand CommandoInsertar(DbConnection conn, CuentaUsuario cuentaUsuario) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "insertarCuentaUsuario";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_userName", DbType.String, cuentaUsuario.UserName);
            this.AgregarParametroEntrada(cmd, "@p_password", DbType.String, cuentaUsuario.Password);
            this.AgregarParametroEntrada(cmd, "@p_activo", DbType.Boolean, cuentaUsuario.IsActive);
            this.AgregarParametroSalida(cmd, "@p_id", DbType.Int32);

            return cmd;
        }

        protected override DbCommand CommandoModificar(DbConnection conn, CuentaUsuario cuentaUsuario) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "modificarCuentaUsuario";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_userName", DbType.String, cuentaUsuario.UserName);
            this.AgregarParametroEntrada(cmd, "@p_password", DbType.String, cuentaUsuario.Password);
            this.AgregarParametroEntrada(cmd, "@p_activo", DbType.Boolean, cuentaUsuario.IsActive);
            this.AgregarParametroEntrada(cmd, "@p_id", DbType.Int32, cuentaUsuario.Id);

            return cmd;
        }

        protected override DbCommand CommandoEliminar(DbConnection conn, int id) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "eliminarCuentaUsuario";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_id", DbType.Int32, id);

            return cmd;
        }

        protected override DbCommand CommandoBuscar(DbConnection conn, int id) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "buscarCuentaUsuarioPorId";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_id", DbType.Int32, id);

            return cmd;
        }

        protected override DbCommand CommandoListar(DbConnection conn) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "listarCuentaUsuarios";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;
            return cmd;
        }

        protected override CuentaUsuario mapearModelo(DbDataReader reader) {
            return new CuentaUsuario {
                Id = Convert.ToInt32(reader["id"]),
                UserName = Convert.ToString(reader["userName"]),
                Password = Convert.ToString(reader["password"]),
                IsActive = (bool)reader["activo"],
            };
        }
    }
}
